codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -74,6 +77,9 @@ public HttpHeaders add(HttpHeaders headers) {
     @Override
     public HttpHeaders set(HttpHeaders headers) {
         if (headers instanceof DefaultHttpHeaders) {
+            if (headers == this) {
+                throw new IllegalArgumentException("can't add to itself.");
```

Comment:
Ok so only for `add(...)` fine by me as well. Not that I think calling set with itself makes any sense though